crates/bevy_core_pipeline/src/prepass/mod.rs

```diff
@@ -0,0 +1,150 @@
+//! Run a prepass before the main pass to generate depth and/or normals texture, also called a g-buffer.
+//! These textures are useful for various screen-space effects and reducing overdraw in the main pass.
+//!
+//! To enable the prepass, you need to add a prepass component to a [`crate::prelude::Camera3d`].
+//!
+//! [`DepthPrepass`]
+//! [`NormalPrepass`]
+//!
+//! Both textures are available on the `PrepassTextures` component attached to each [`crate::prelude::Camera3d`] with a prepass component.
+//! The textures are automatically added to the default mesh view bindings. You can also get the raw textures
+//! by querying the [`ViewPrepassTextures`] component on any camera with a prepass component.
+//!
+//! When using the default mesh view bindings you should be able to use `prepass_depth()`
+//! and `prepass_normal()` to load the related textures. These functions are defined in `bevy_pbr::utils`.
+//! See the `shader_prepass` example that shows how to use it.
+//!
+//! The prepass runs for each `Material`, you can control if the prepass should run by setting the `prepass_enabled`
+//! flag on the `MaterialPlugin`.
+//!
+//! Currently only works for 3d
+
+pub mod node;
+
+use bevy_ecs::prelude::*;
+use bevy_reflect::Reflect;
+use bevy_render::{
+    render_phase::{CachedRenderPipelinePhaseItem, DrawFunctionId, EntityPhaseItem, PhaseItem},
+    render_resource::{CachedRenderPipelineId, Extent3d, TextureFormat},
+    texture::CachedTexture,
+};
+use bevy_utils::FloatOrd;
+
+pub const DEPTH_PREPASS_FORMAT: TextureFormat = TextureFormat::Depth32Float;
+pub const NORMAL_PREPASS_FORMAT: TextureFormat = TextureFormat::Rgb10a2Unorm;
+
+/// If added to a [`crate::prelude::Camera3d`] then depth values will be copied to a separate texture available to the main pass.
+/// The main pass already uses a depth texture by default which helps reduce overdraw, but this will help reduce it even more.
```

Comment:
this is outdated now - the copied depth texture doesn't affect overdraw reduction

Reply:
That's not true? The depth texture is used in the main pass and uses the GreaterEqual comparison operator. So only fragments with the same depth value or greater compared to the value in the depth buffer will be shaded. That reduces overdraw compared to shading each time a fragment is closer than the previous closest, as the prepass finds _the_ closest.

Reply:
i'm not sure what the right terminology should be, but i am going with "depth buffer" for the thing that gets attached to the prepass and is filled with depths (whatever prepass outputs are used), and "depth texture" for the bindable copy that gets made if the `DepthPrepass` component is added.

the depth buffer should be used in the main pass whatever prepass output is requested (and maybe we should also have a `JustRunThePrepass` component as well that generates no explicit output, if you just want the overdraw reduction).

the depth texture is a bindable copy of the depth buffer we make because you can't bind and attach it in the main pass at the same time. whether we make the copy shouldn't affect whether we load the depth buffer in the main pass or not.

Reply:
Yeah, I meant depth buffer here, but you are right that it indeed doesn't matter anymore since it's always used even without this component.

```suggestion

```

I kinda like the idea of a `JustRunThePrepass` or `OverdrawPrepass` maybe, but I think I'd rather keep it simple for now and do it in a follow up PR.

Reply:
looks like this is still present

Reply:
Are you sure @robtfm? I'm not seeing it locally

Reply:
sorry, you're totally right... i clicked the file link on my comment above and apparently it took me to the version at the time instead of the current one.